galasa-ui/src/tests/routes/users.test.ts

```diff
@@ -7,12 +7,43 @@
 import { UsersAPIApi } from '@/generated/galasaapi';
 import { DELETE, GET } from '../../app/users/route';
 import { createAuthenticatedApiConfiguration } from '../../utils/api';
-import * as Constants from "@/utils/constants";
+import * as Constants from '@/utils/constants';
 import AuthCookies from '@/utils/authCookies';
+import { NextResponse } from 'next/server';
 
+// Mock modules and dependencies
 jest.mock('../../utils/api');
 jest.mock('@/generated/galasaapi');
 
+// Mock NextResponse to handle both static and instance methods
+jest.mock('next/server', () => {
+  const actualNextServer = jest.requireActual<typeof import('next/server')>('next/server');
+
+  class MockNextResponse {
+    body: any;
+    status: number;
+    headers: Headers;
+
+    constructor(body: any, init?: ResponseInit) {
+      this.body = body;
+      this.status = init?.status || 200;
+      this.headers = new Headers(init?.headers);
+    }
+
+    static json(data: any, init?: ResponseInit) {
+      return new MockNextResponse(JSON.stringify(data), init);
+    }
+
+    async json() {
+      return JSON.parse(this.body);
+    }
+  }
+
+  return {
+    ...actualNextServer,
+    NextResponse: MockNextResponse,
+  };
```

Comment:
Is this needed? We already have tests that use NextResponse without needing to mock it out